File: week2/윤여준/로또의 최고 순위와 최저 순위.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer(2);
    
    set<int> nums;
    for(int i : win_nums) nums.insert(i);
    
    int cnt = 0, bonus = 0;
    for(int i : lottos) {
        if(i == 0) {
            bonus++;
            continue;
        }
        if(nums.find(i) != nums.end()) {
            cnt++;
            nums.erase(i);
        }
    }
    
    answer[0] = min(7 - cnt - bonus, 6);
    answer[1] = min(7 - cnt, 6);
    
    return answer;
}

File: week2/이지훈/로또의 최고 순위와 최저 순위.java
public class 로또의 최고 순위와 최저 순위 {
    public int[] solution(int[] lottos, int[] win_nums) {
        int matchCnt = 0;
        int zeroCnt = 0;
        int[] check = new int[46];

        for (int i = 0; i < 6; i++) {
            check[lottos[i]]++;
            check[win_nums[i]]++;
        }

        for (int i = 1; i < 46; i++) {
            if (check[i] == 2) matchCnt++;
        }
        zeroCnt = check[0];

        int[] result = new int[2];
        result[0] = Math.min(7 - matchCnt + zeroCnt, 6); // 최고순위
        result[1] = Math.min(7 - matchCnt, 6); // 최저순위

        return result;
    }

}
